src/averageAges.js

```diff
@@ -55,7 +69,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredPeople = onlyWithSon
+    ? people.filter(({ sex, mother }) => (
+      people.find(({ name }) => mother === name)) && sex === 'm')
+    : people.filter(({ mother }) => (
+      people.find(({ name }) => mother === name)));
```

Comment:
Method `some` would be better use for this purpose then `find`, because this way as I understand you need only boolean, not entire person object.